### DISCUSSION TITLE
How to get more than 3-5 Mbps with external WiFi AP and Wireguard?

### DISCUSSION THREAD

**Discussion by asmod3us** [RaspAP/raspap-webgui#1131] [Q&A]
Hello,

I've built a travel router with RaspAP on RPI4 with external WiFi Dongle (Alfa AWUS036ACM).
It's connecting to a WiFi network with the onboard WiFi adapter and uses the dongle for the hotspot.

This currently runs at a speed of 3-5 Mbps with a Wireguard tunnel. Can this go any faster?

I've tested a GL-iNet MT1300 (Beryl) that does 70-80 Mbps with the same Wireguard tunnel, so I'm confident there is some potential for tuning, but I don't know where to start. Have you got any ideas?

Thank you,
Achim

**Comment by billz** [comment]
Identifying network bottlenecks is both an art and science. A good approach in this case is to measure throughput with `iperf` between your Pi4's onboard chipset and another machine on your local network. Perform the test with and without the tunnel. Repeat the test with the external dongle (again, with and without the tunnel) and tabulate the results. 

WireGuard does not have many settings that can be tuned. The Pi4 usually tops out at 800-900 Mbps when using WG on a local wired network. Your poor throughput can be the result of any number of factors, such as interference on the AC band (are your WiFi adapters using different AC channels?), under voltage, etc. 

Bear in mind that the Beryl is built on ASICs, whereas the Pi is a general purpose SBC. So this is comparing apples to oranges.

I haven't personally done any benchmarking with WG in your configuration, so perhaps others can chime in.
